Foundation/NSURLSession/NSURLSessionTask.swift

```diff
@@ -597,10 +618,11 @@ fileprivate extension URLSessionTask {
     /// expects.
     ///
     /// - SeeAlso: https://curl.haxx.se/libcurl/c/CURLOPT_HTTPHEADER.html
-    func curlHeaders(for request: URLRequest) -> [String] {
+    func curlHeaders(for httpHeaders: [AnyHashable : Any]?) -> [String] {
         var result: [String] = []
         var names = Set<String>()
-        if let hh = currentRequest?.allHTTPHeaderFields {
+        if httpHeaders != nil {
+            let hh = httpHeaders as! [String:String]
```

Comment:
@naithar I agree with you that String(describing:) is the preferred way to convert an instance of any type to a string as per the documentation. I think it makes sense if we need to convert only specific elements in the dictionary. Here, we need to process each element in the dictionary and hence converting the dictionary from any type to a string dictionary makes the processing simpler.

We cannot use .lowercased() and .isEmpty methods with the elements of the [AnyHashable : Any] dictionary. Hence, we need to convert them to string before applying those methods.

In order to process the httpHeaders as any type, we need to make following changes.
        var names = Set<String>()
        if httpHeaders != nil {
           // let hh = httpHeaders as! [String:String]
            httpHeaders?.forEach {
                let name = String(describing:$0.0).lowercased()
                guard !names.contains(name) else { return }
                names.insert(name)
                
                if String(describing:$0.1).isEmpty {
                    result.append(String(describing:$0.0) + ";")
                } else {
                    result.append(String(describing:$0.0) + ": " + String(describing:$0.1))
                }
            }
        }
As the element values are already converting to string, I did not change the names variable.

Please let me know your thoughts whether the existing changes are fine or you have any better way.